### PR TITLE
Fix ingress template for GKE comptibility

### PR DESCRIPTION
**Pull request by Mofef**
I had problems analog to the issue linked below and the same solution worked for me aswell. I hope this does not conflict with any other implementation.
https://github.com/jetstack/kube-lego/issues/115#issuecomment-290235071

### CODE REVIEW

**Comment by mouradmourafiq** [comment]
Hi thanks for the PR, I have a question, does the cli logs stream correctly?

**Comment by mouradmourafiq** [comment]
Just read the issue you linked, and it should work, not sure if this will break nginx though.

**Comment by lgeiger** [comment]
> Hi thanks for the PR, I have a question, does the cli logs stream correctly?

I am using the same config on NGINX and `polyaxon experiment -xp 4930 logs` or `polyaxon experiment -xp 4930 logs --follow` doesn't work. Only `polyaxon experiment -xp 4930 logs --past` works. Though I am not sure if that's related.

**Comment by Mofef** [comment]
Thx for checking that @lgeiger, I didn't find the time yet.

**Comment by mouradmourafiq** [comment]
@lgeiger that's exactly what I wanted to know.

Seems like this is only for GKE ingress, why do you need that for vanilla nginx ingress? I am sure I tested the current config with an nginx ingress from stable helm chart and it works.

**Comment by lgeiger** [comment]
> Seems like this is only for GKE ingress, why do you need that for vanilla nginx ingress? I am sure I tested the current config with an nginx ingress from stable helm chart and it works.

When I setup the GKE ingress, Polyaxon didn't load the CSS assets. This seamed to fix it.
I can try to remove this "fix" next week and we can see if it still works (We have a paper deadline coming up so I won't play with our "production" deployment right now).

**Comment by lgeiger** [comment]
I removed the fix by editing the ingress deployment and the polyaxon UI seams fine 🎉 

Unfortunately `polyaxon experiment logs` still fails with the following error:
```shell
Setting OrderedDict([('check_count', 12), ('current_version', '0.4.4'), ('min_version', '0.4.2'), ('log_handler', OrderedDict([('dsn', 'aHR0cHM6Ly84MTc4MjEwYjY4NDU0OGE1Yjk2YjgxMzFmYTY0NWM5ZTozY2IwZGI1OTY5NDU0NjNkYTUzMmUzNmRiZTQ4NjFmNUBzZW50cnkuaW8vMTE5NzQxNg=='), ('environment', 'production'), ('tags', {'cli_min_version': '0.4.2', 'cli_latest_version': '0.4.4', 'platform_min_version': '0.4.2', 'platform_latest_version': '0.4.4', 'chart_version': '0.4.4'})]))]) in the file .polyaxoncli
Starting request to url: https://polyaxon.mydomain.com:443/api/v1/versions/cli/ with params: None, data: None
Starting new HTTPS connection (1): polyaxon.mydomain.com:443
https://polyaxon.mydomain.com:443 "GET /api/v1/versions/cli/ HTTP/1.1" 200 48
Response Content: b'{"latest_version":"0.4.4","min_version":"0.4.2"}', Headers: {'Server': 'nginx/1.10.3', 'Date': 'Thu, 30 May 2019 20:13:53 GMT', 'Content-Type': 'application/json', 'Content-Length': '48', 'Allow': 'GET, HEAD, OPTIONS', 'X-Frame-Options': 'SAMEORIGIN', 'Vary': 'Origin, Cookie', 'Via': '1.1 google', 'Alt-Svc': 'clear'}
Setting OrderedDict([('check_count', 12), ('current_version', '0.4.4'), ('min_version', '0.4.2'), ('log_handler', OrderedDict([('dsn', 'aHR0cHM6Ly84MTc4MjEwYjY4NDU0OGE1Yjk2YjgxMzFmYTY0NWM5ZTozY2IwZGI1OTY5NDU0NjNkYTUzMmUzNmRiZTQ4NjFmNUBzZW50cnkuaW8vMTE5NzQxNg=='), ('environment', 'production'), ('tags', {'cli_min_version': '0.4.2', 'cli_latest_version': '0.4.4', 'platform_min_version': '0.4.2', 'platform_latest_version': '0.4.4', 'chart_version': '0.4.4'})]))]) in the file .polyaxoncli
Termination cause: Handshake status 404 Not Found
Session disconnected.
Session ended
```
Should I open a new issue for this?

**Comment by mouradmourafiq** [comment]
next release there will be one port exposed, so it should be fixed in that case.

**Comment by Mofef** [comment]
In the meanwhile I could reproduce the issue (logs don't get streamed). Will try nginx ingress next. Anyway I think we can close this.

**Comment by Mofef** [comment]
Sadly, after changing to nginx ingress, log streaming is still not working for me. But I don't need to edit the path and the webinterface looks ok. :tada: 

I'm not sure if this could be related but I get SSL verification warnings (see below) with every API call (because verification is off. If I enable it I get verification errors. feel free to check: https://polyaxon.magazino.eu )

@mouradmourafiq If this is not related I would open two separate issues. (here in the chart repo I guess?)

Example of SSL warnings and failing log stream
```
$ polyaxon run -u -l --ttl 120 
/home/muenst/polyaxon_env/local/lib/python2.7/site-packages/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
[================================] 8145/8145 - 00:00:00
Files uploaded.
Creating an independent experiment.
/home/muenst/polyaxon_env/local/lib/python2.7/site-packages/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
Experiment `66` was created
Session ended
```

This is the generated ingress
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
  creationTimestamp: 2019-06-17T13:25:11Z
  generation: 1
  labels:
    app.kubernetes.io/component: polyaxon-api
    app.kubernetes.io/instance: polyaxon-0.4.4
    app.kubernetes.io/managed-by: helm
    app.kubernetes.io/name: polyaxon-polyaxon-ingress
    app.kubernetes.io/part-of: polyaxon-core
    app.kubernetes.io/version: 0.4.4
    chart: polyaxon-0.4.4
    heritage: Tiller
    release: polyaxon
    role: polyaxon-api
    type: polyaxon-core
  name: polyaxon-polyaxon-ingress
  namespace: polyaxon
  resourceVersion: "12437542"
  selfLink: /apis/extensions/v1beta1/namespaces/polyaxon/ingresses/polyaxon-polyaxon-ingress
  uid: 558c4f2c-9103-11e9-9b76-42010aa40008
spec:
  rules:
  - host: polyaxon.magazino.eu
    http:
      paths:
      - backend:
          serviceName: polyaxon-polyaxon-api
          servicePort: 80
        path: /
      - backend:
          serviceName: polyaxon-polyaxon-api
          servicePort: 1337
        path: /ws
  tls:
  - hosts:
    - polyaxon.magazino.eu
    secretName: polyaxon-cert
status:
  loadBalancer:
    ingress:
    - ip: 35.204.139.217
```

**Comment by mouradmourafiq** [comment]
I think the easiest workaround here to allow people to customize the path is to update the PR with:
```
```

File: polyaxon/templates/ing.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
   - host: {{ .Values.ingress.hostName | quote }}
     http:
       paths:
-      - path: /
+      - path: /*
```
Comment (mouradmourafiq) [inline polyaxon/templates/ing.yaml]:
```suggestion
      - path: {{ default "/" .Values.ingress.path | quote }}
```

**Comment by mouradmourafiq** [comment]
I am closing this PR, I will push the change myself for the next release. Thank you for the explanation and for creating the PR.

**Comment by Mofef** [comment]
Thank you! Sorry for the late response... I must have missed the notification.
Meanwhile, I was busy with other tasks while our installation of 0.4.4 was running very well and creepingly transitioned from a test installation to a loved tool used in everyday development. This week I'm going to attempt the migration to 0.5.x and I'm very much looking forward to it :)

**Pull request closed without merging**